### ISSUE TITLE
Secret management hitting maximum value size

### ISSUE THREAD

**Issue by superseb** [rancher/rancher#7708]
**Rancher Versions:**
Server: 1.4.0
healthcheck: v0.2.3
ipsec: 0.0.4
network-services: v0.0.10
scheduler: v0.4.0
secrets: v0.0.1
kubernetes (if applicable):

**Docker Version:**
Docker 1.12.6
**OS and where are the hosts located? (cloud, bare metal, etc):**
Ubuntu 16.04 on AWS
**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
Single node rancher, with 1 agent
**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Cattle
**Steps to Reproduce:**

1) Launch Rancher/server
2) Launch Rancher secrets from the catalog
3) Add host to server
4) Add secret with key = test, value anything bigger than 141 bytes. For example 'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'

**Results:** 
Container is launched but volume with secret is not mounted.

Logging rancher/server:
```
time="2017-02-06T20:02:01Z" level=error msg="Could not decrypt secret"
time="2017-02-06T20:02:01Z" level=error msg="crypto/rsa: message too long for RSA public key size"
time="2017-02-06T20:02:01Z" level=error msg="Error in request, code : 400: crypto/rsa: message too long for RSA public key size"
```

Logging secrets container:
```
06/02/2017 21:46:17time="2017-02-06T20:46:17Z" level=info msg=create.request name=ea367a4f41c10362bebfa8a34a0b0a57dc0b00517418d45bfc8017d052dbe19c options=map[io.rancher.secrets.token:{"value":"eyJhbGciOiJSUzI1NiJ9.eyJzdWIiOiJjYXR0bGUiLCJraWQiOiJkZWZhdWx0IiwiaXNzIjoiaHR0cDpcL1wvY2F0dGxlLmlvIiwiZXhwIjoxNTE3OTcwOTAzLCJ1dWlkIjoiNTg5ODNiNmYtM2ZhZi00ZTBjLTk3MzYtMjQ4NzQ4MjMyNmQ1IiwiaWF0IjoxNDg2NDEzOTc3fQ.lYMjBmSE3rhzb_DXmaU-aJSRYVATN8hBxx3yxoxQe9TSH75SjzUs4j7KCLIgsyXTaxNI3lJqrP2H8PGiyoP7i5W-Up0wIPwD8i7EZEjlcV6g7HkiC1Msia-6tXZfugik1A4-cW4XAzP4CxDNNzQuYXxwlTXxhO6e59HrU0C-AXIP2hrcCPGIew39H8bp4vqicV4wHCLMyKf47uxqcOaShIZwc-Lf3kA4DsgtPraXlU7oSkPlUhZv1xiiAG05eTU7COELbfJbPkPeDcePTYL791O4GxAQ7wdBn6jkLJWP0Ttvy6Rf1clLpAaqC5IazNEzHPUWj982uSBTdmS4SoqJQg"}]
06/02/2017 21:46:17time="2017-02-06T20:46:17Z" level=info msg="Updating volume ea367a4f41c10362bebfa8a34a0b0a57dc0b00517418d45bfc8017d052dbe19c (1v124) rancher-secrets:1sd1 on 1h2, state inactive => inactive" err=<nil>
06/02/2017 21:46:17time="2017-02-06T20:46:17Z" level=info msg=create.response
06/02/2017 21:46:17time="2017-02-06T20:46:17Z" level=info msg=attach.request name=ea367a4f41c10362bebfa8a34a0b0a57dc0b00517418d45bfc8017d052dbe19c
06/02/2017 21:46:17time="2017-02-06T20:46:17Z" level=error msg="Unsuccessful request: 500 Internal Server Error"
06/02/2017 21:46:17time="2017-02-06T20:46:17Z" level=info msg=attach.response
06/02/2017 21:46:17time="2017-02-06T20:46:17Z" level=info msg=mount.request name=ea367a4f41c10362bebfa8a34a0b0a57dc0b00517418d45bfc8017d052dbe19c
06/02/2017 21:46:17time="2017-02-06T20:46:17Z" level=error msg="Unsuccessful request: 500 Internal Server Error"
06/02/2017 21:46:17time="2017-02-06T20:46:17Z" level=info msg=mount.response mountpoint="/var/lib/rancher/volumes/rancher-secrets/ea367a4f41c10362bebfa8a34a0b0a57dc0b00517418d45bfc8017d052dbe19c"
06/02/2017 21:49:19time="2017-02-06T20:49:19Z" level=info msg=remove.request name=3f460ea0465e3da8fbbe9407908423399d83e7901dcdf99d5c90b62bf7961173
06/02/2017 21:49:19time="2017-02-06T20:49:19Z" level=info msg=remove.response
06/02/2017 21:49:19time="2017-02-06T20:49:19Z" level=info msg=remove.request name=0dda41d1971a25932b941189b47ea6b7b59abdf0fb1a1be2964bc5a2b4d51498
06/02/2017 21:49:19time="2017-02-06T20:49:19Z" level=info msg=remove.response
```

**Comment by cloudnautique** [comment]
The fixes will be in 1.5.0-rc1.

You should now be able to store larger items such as private keys. 

To test this:
1. Create secrets in 1.4.x both large/small.
2. Deploy containers (this is initial functionality, and should fail with large secrets)
2. Upgrade Rancher to 1.5.0-rcX
2. Upgrade secrets volume driver
2. Deploy containers (Both large and small secrets should be created)

**Comment by cloudnautique** [comment]
1.5.0-rc1 is out now

**Comment by galal-hussein** [comment]
I followed the same steps:

1. Create secrets in 1.4.x both large/small.
2. Deploy containers (this is initial functionality, and should fail with large secrets)
3. Upgrade Rancher to 1.5.0-rcX
4. Upgrade secrets volume driver
5. Deploy containers (Both large and small secrets should be created)

I was able to deploy the containers with mounted secrets after the upgrade from v1.4.1 to v.1.5.0-rc1.
